app/src/app.js

```diff
@@ -5,10 +5,62 @@ const app = express();
 const bodyParser = require('body-parser');
 app.use(bodyParser.json());
 
-app.get('/posts', async(req, res)=>{
-            let posts = await postModel.getPosts();
-            res.json(posts);
-        });
+
+app.put('/posts/:id', async (req, res) => {
+
+    try {
+        if (isNaN(req.params.id)) {
+            res.status(400).json({
+                message: 'id must be a number'
+            });
+            return;
+        }
+        let postById = await postModel.getPost(req.params.id)
+        if (postById === null) {
+            res.status(404);
+            res.end();
+            return;
+        }
+        postById.title = await req.body.title;
+        postById.slug = await req.body.slug;
+        postById.content = await req.body.content;
+        
+        const error = [];
+        if (typeof postById.title !== 'string') {
```

Comment:
Mamy tutaj spory kawałek powielonego kodu do walidacji. Jeśli dojdzie nam warunek że np. `slug` nie może być dłuższy niż 50 znaków to będzie trzeba przerabiać walidację w wielu miejscach. Dodaj plik `app/src/validation/post.js` i tam funkcję `validatePost` która będzie przyjmowała `PostEntity` a zwracała listę błędów. Nie zapomnij o testach ;P